core/server/api/v2/integrations.js

```diff
@@ -0,0 +1,151 @@
+        options: [
+            'include',
+            'limit'
+        ],
+        validation: {
+            options: {
+                include: {
+                    values: ['api_keys', 'webhooks']
+                }
+            }
+        },
+        query({options}) {
+            return models.Integration.findPage(options);
+        }
+    },
+    read: {
+        permissions: true,
+        data: [
+            'id'
+        ],
+        options: [
+            'include'
+        ],
+        validation: {
+            data: {
+                id: {
+                    required: true
+                }
+            },
+            options: {
+                include: {
+                    values: ['api_keys', 'webhooks']
+                }
+            }
+        },
+        query({data, options}) {
+            return models.Integration.findOne(data, Object.assign(options, {require: true}))
+                .catch(models.Integration.NotFoundError, () => {
+                    throw new common.errors.NotFoundError({
+                        message: common.i18n.t('errors.api.resource.resourceNotFound', {
+                            resource: 'Integration'
+                        })
+                    });
+                });
+        }
+    },
+    edit: {
+        permissions: true,
+        data: [
+            'name',
+            'icon_image',
+            'description',
+            'webhooks'
+        ],
+        options: [
+            'id',
+            'include'
+        ],
+        validation: {
+            options: {
+                id: {
+                    required: true
+                },
+                include: {
+                    values: ['api_keys', 'webhooks']
+                }
+            }
+        },
+        query({data, options}) {
+            return models.Integration.edit(data, Object.assign(options, {require: true}))
+                .catch(models.Integration.NotFoundError, () => {
+                    throw new common.errors.NotFoundError({
+                        message: common.i18n.t('errors.api.resource.resourceNotFound', {
+                            resource: 'Integration'
+                        })
+                    });
+                })
+                .then((model) => {
+                    return model.fetch(options);
+                });
+        }
+    },
+    add: {
+        permissions: true,
+        data: [
+            'name',
+            'icon_image',
+            'description',
+            'webhooks'
+        ],
+        options: [
+            'include'
+        ],
+        validation: {
+            data: {
+                name: {
+                    required: true
+                }
+            },
+            options: {
+                include: {
+                    values: ['api_keys', 'webhooks']
+                }
+            }
+        },
+        query({data, options}) {
+            return models.Base.transaction((transacting) => {
```

Comment:
IMO transaction should be created inside the model

Reply:
Hmm okay - would we do that regardless of whether `api_keys` or `webhooks` was passed in? Or only if they are?

Reply:
for `edit` regardless, because we do a fetch + save -> we lock the row.
for `add` it could be dependent on the passed in relations, but i would not go that far?

You can look at the post model implementation.
The transaction handling could live in the base model, but nothing we can refactor now.

Reply:
Okay that sounds good - you're deffo right about it being in the model :relaxed: 

>  The transaction handling could live in the base model, but nothing we can refactor now.

Yeah maybe even better - another time tho...